### PR TITLE
Global switch_version attribute created and used by telemetryStatsServerP check

### PR DESCRIPTION
**Pull request by wilsonbc2**
```
< 13:44:34 benjwils ./ACI-Pre-Upgrade-Validation-Script/tests/telemetryStatsServerP_object_check> $ pytest -v test_telemetryStatsServerP_object_check.py
====================================================================================================================================== test session starts =======================================================================================================================================
platform darwin -- Python 2.7.18, pytest-4.6.11, py-1.11.0, pluggy-0.13.1 -- /Library/Frameworks/Python.framework/Versions/2.7/Resources/Python.app/Contents/MacOS/Python
cachedir: .pytest_cache
rootdir: ./ACI-Pre-Upgrade-Validation-Script, inifile: pytest.ini
collected 10 items

test_telemetryStatsServerP_object_check.py::test_logic[icurl_outputs0-4.2(1b)-5.2(2a)-PASS] PASSED                                                                                                                                                                                         [ 10%]
test_telemetryStatsServerP_object_check.py::test_logic[icurl_outputs1-3.2(1a)-4.2(4d)-PASS] PASSED                                                                                                                                                                                         [ 20%]
test_telemetryStatsServerP_object_check.py::test_logic[icurl_outputs2-3.2(1a)-5.2(6a)-PASS] PASSED                                                                                                                                                                                         [ 30%]
test_telemetryStatsServerP_object_check.py::test_logic[icurl_outputs3-4.2(3a)-4.2(7d)-PASS] PASSED                                                                                                                                                                                         [ 40%]
test_telemetryStatsServerP_object_check.py::test_logic[icurl_outputs4-2.2(3a)-2.2(4r)-PASS] PASSED                                                                                                                                                                                         [ 50%]
test_telemetryStatsServerP_object_check.py::test_logic[icurl_outputs5-5.2(1a)-None-MANUAL CHECK REQUIRED] PASSED                                                                                                                                                                           [ 60%]
test_telemetryStatsServerP_object_check.py::test_logic[icurl_outputs6-4.2(1b)-5.2(2a)-PASS] PASSED                                                                                                                                                                                         [ 70%]
test_telemetryStatsServerP_object_check.py::test_logic[icurl_outputs7-3.2(1a)-4.2(4d)-PASS] PASSED                                                                                                                                                                                         [ 80%]
test_telemetryStatsServerP_object_check.py::test_logic[icurl_outputs8-3.2(1a)-5.2(6a)-FAIL - OUTAGE WARNING!!] PASSED                                                                                                                                                                      [ 90%]
test_telemetryStatsServerP_object_check.py::test_logic[icurl_outputs9-4.2(3a)-4.2(7d)-PASS] PASSED                                                                                                                                                                                         [100%]

=================================================================================================================================== 10 passed in 0.10 seconds ====================================================================================================================================


// Fabric has four switch versions -- lowest version is selected
apic1# moquery -c firmwareRunning | grep "peVer" | sort | uniq
peVer            : 5.2(6e)
peVer            : 5.2(8g)
peVer            : 6.0(2.200)
peVer            : 6.0(3e)

apic1# python upgrade.py
--- omit ---
Gathering Switch Versions from Firmware Repository...5.2(6e)
--- omit ---
```

### CODE REVIEW

File: aci-preupgrade-validation-script.py [hunk 1]
```diff
@@ -2862,7 +2882,7 @@ def access_untagged_check(index, total_checks, **kwargs):
         sys.exit()
     inputs = {'username': username, 'password': password,
               'cversion': cversion, 'tversion': tversion,
-              'vpc_node_ids': vpc_nodes}
+              'vpc_node_ids': vpc_nodes, 'switch_version':switch_version}
```
Comment (monrog2) [inline aci-preupgrade-validation-script.py:2885]: thoughts on changing the common vars to be more clear:
`cversion > apic_cversion`
`switch_version > sw_cversion`
`tversion > fab_tversion` (this one I could go without)

File: aci-preupgrade-validation-script.py [hunk 2]
```diff
@@ -691,6 +691,25 @@ def get_vpc_nodes(**kwargs):
 
     return vpc_nodes
 
+def get_switch_version(**kwargs):
+    """ Returns: AciVersion instance """
```
Comment (monrog2) [inline aci-preupgrade-validation-script.py:695]: can be clear that we're returning the lowest found sw version

File: aci-preupgrade-validation-script.py [hunk 3]
```diff
@@ -691,6 +691,25 @@ def get_vpc_nodes(**kwargs):
...
+    prints("Gathering Switch Versions from Firmware Repository...", end='')
+    firmwares = icurl('class', 'firmwareRunning.json')
+    versions = []
+    switch_version = None
```
Comment (monrog2) [inline aci-preupgrade-validation-script.py:699]: consider renaming this lowest_switch_version or something along those lines

File: aci-preupgrade-validation-script.py [hunk 4]
```diff
@@ -691,6 +691,25 @@ def get_vpc_nodes(**kwargs):
...
+
+    for firmware in firmwares:
+        if firmware['firmwareRunning']['attributes']['peVer'] not in versions:
```
Comment (monrog2) [inline aci-preupgrade-validation-script.py:702]: minor nitpick, can use a set here to force uniqueness

File: aci-preupgrade-validation-script.py [hunk 5]
```diff
@@ -691,6 +691,25 @@ def get_vpc_nodes(**kwargs):
...
+            versions.append(firmware['firmwareRunning']['attributes']['peVer'])
+    for version in versions:
+        version = AciVersion(version)
+        if not switch_version:
```
Comment (monrog2) [inline aci-preupgrade-validation-script.py:706]: think you can collapse this further by popping the first val to `switch_version` then iterating through the remaining:
```
    versions = set()
    try:
        lowest_sw_ver = AciVersion(versions.pop())
    except KeyError:
        #Error, did not find any switch versions
        
    for ver in versions:
        if lowest_sw_ver.newer_than(str(ver)):
            lowest_sw_ver = AciVersion(ver)

**Review by monrog2 — CHANGES_REQUESTED** [review]
Lets chat about the global var names given the sw add

File: aci-preupgrade-validation-script.py (hunk 1 above)
Comment (wilsonbc2) [reply aci-preupgrade-validation-script.py]: Agree. Changed switch global var to sw_cversion for now. We can open up a separate enhancement to change cversion > apic_cversion once we get through the open issues.

**Comment by wilsonbc2** [comment]
Pushed changes based on Gabe's comments.

**Review by monrog2 — APPROVED** [review]
all good, will file a separate issue to change the current `cversion` to `apic_cversion` for consistency sake.

**Pull request merged**